go-controller/pkg/metrics/master.go

```diff
@@ -843,7 +843,12 @@ type ConfigDurationRecorder struct {
 // global variable is needed because this functionality is accessed in many functions
 var cdr *ConfigDurationRecorder
 
+// lock for accessing the cdr global variable
+var cdrMutex sync.RWMutex
```

Comment:
why not a normal mutex, why RW?

Reply:
didn't think too much, lol let me change to normal mutex since that is more perf efficient